async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+				}
+
+			} catch (KafkaException e) {
+				
+				producer.abortTransaction();
+
+			}
+
+		}
+	}
+
+	private static KafkaConsumer<String, String> createKafkaConsumer() {
+    	Properties props = new Properties();
+    	props.put(BOOTSTRAP_SERVERS_CONFIG, "asynckafka1:9092");
+    	props.put(GROUP_ID_CONFIG, CONSUMER_GROUP_ID);
+    	props.put(ENABLE_AUTO_COMMIT_CONFIG, "false");
+    	props.put(ISOLATION_LEVEL_CONFIG, "read_committed");
+		props.put(MAX_POLL_RECORDS_CONFIG, 1);
+        props.put(AUTO_OFFSET_RESET_CONFIG, "earliest");
+    	props.put(KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+    	props.put(VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+
+    	KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props);
+    	consumer.subscribe(singleton(INPUT_TOPIC));
+    	return consumer;
+	}
+
+	private static KafkaProducer<String, String> createKafkaProducer() {
+
+    	Properties props = new Properties();
+    	props.put(BOOTSTRAP_SERVERS_CONFIG, "asynckafka1:9092");
+    	props.put(ENABLE_IDEMPOTENCE_CONFIG, "true");
+    	props.put(TRANSACTIONAL_ID_CONFIG, "prod-1");
+    	props.put(KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    	props.put(VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+
+    	return new KafkaProducer(props);
+
+	}
+	
+	private Response ExecDestinationAPIMain(KafkaConsumer<String, String> consumer, KafkaProducer<String, String> producer) {
+
+		StringBuilder destinationStatus = new StringBuilder();
+		StringBuilder destinationResponseBody = new StringBuilder();
+		String receiveMessage = new String();
+		StringBuilder executionTimes = new StringBuilder();
+		StringBuilder sendMessage = new StringBuilder();
+		StringBuilder retryMessage = new StringBuilder();
+		Responses response = new Responses();
+
+		receiveMessage = null;
+		destinationStatus.delete(0, destinationStatus.length());
+		destinationResponseBody.delete(0, destinationResponseBody.length());
+		executionTimes.delete(0, executionTimes.length());
+		sendMessage.delete(0, sendMessage.length());
+		retryMessage.delete(0, retryMessage.length());
+		consumer.subscribe(Arrays.asList(INPUT_TOPIC));
+
+		// メッセージをTopic1から読む
+		ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(60l));
+
+		for (ConsumerRecord<String, String> record : records) {
+			System.out.println(String.format("*record* %s:%s", record.offset(), record.value()));
+			receiveMessage = record.value();
+		}
+
+		// メッセージが拾えなかったら
+		if(receiveMessage == null) {
+			System.out.println("Async4BackendRest_DestinationAPI_Topic relode");
+			//continue;
+			return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+					}
+
+		Response typeresponse = ExecuteDestinationAPI(receiveMessage, destinationStatus, destinationResponseBody, executionTimes);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *typeresponse*. Execution paths that do not contain closure statements were detected. Ensure *typeresponse* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).